### PR TITLE
Integration tests: fix flaky tests `test_backup_restore_on_cluster/test_concurrency.py` & `test_manipulate_statistics/test.py`

### PR DESCRIPTION
**Pull request by fm4v**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [x]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [x]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: tests/integration/test_backup_restore_on_cluster/test_concurrency.py [hunk 1]
```diff
@@ -276,15 +276,10 @@ def make_backups():
...
         assert_eq_with_retry(
             node,
-            f"SELECT status from system.backups WHERE id IN {ids_list} AND (status == 'CREATING_BACKUP')",
-            "",
-        )
-
-    for node in nodes:
-        assert_eq_with_retry(
-            node,
-            f"SELECT status, error from system.backups WHERE id IN {ids_list} AND (status == 'BACKUP_FAILED')",
```
Comment (vitlibar) [inline tests/integration/test_backup_restore_on_cluster/test_concurrency.py:286]: Why was it removed? We must check that backups were successful.

Comment (fm4v) [reply tests/integration/test_backup_restore_on_cluster/test_concurrency.py:286]: It is moved to single query
https://github.com/ClickHouse/ClickHouse/pull/67027/files#diff-b0c6059a60439150d892d295856de7dcd042339e92efbf8df8276419f6791f1dR280

Comment (vitlibar) [reply tests/integration/test_backup_restore_on_cluster/test_concurrency.py:286]: When a backup fails it's better to be able to see in the test results why it happened.

Your code won't show us any errors, whereas the removed code could do that:
```
assert_eq_with_retry(
            node,
            f"SELECT status, error from system.backups WHERE id IN {ids_list} AND (status == 'BACKUP_FAILED')",
    "")
```

And your code won't show

Comment (fm4v) [reply tests/integration/test_backup_restore_on_cluster/test_concurrency.py:286]: You are right about this

**Comment by fm4v** [comment]
I also made small typo fix for setting `throw_if_deduplication_in_dependent_materialized_views_enabled_with_async_insert` in this PR
https://github.com/ClickHouse/ClickHouse/pull/67027/commits/09cb2d822a0960f9917f28d36c47c1e67e861178

Don't want to run CI twice

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 09cb2d822a0960f9917f28d36c47c1e67e861178 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67027/09cb2d822a0960f9917f28d36c47c1e67e861178/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67027/09cb2d822a0960f9917f28d36c47c1e67e861178/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Pull request merged**